modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -232,47 +251,32 @@ export default {
       this.$emit('esc');
     },
     trapFocus () {
-      // Adapted from https://uxdesign.cc/how-to-trap-focus-inside-modal-to-make-it-ada-compliant-6a50f9a70700
-      // All the elements inside modal which you want to make focusable.
-      const focusableElements = [
-        'button',
+      const elementSelectors = [
+        '[tabindex]',
         '[href]',
         'input',
         'select',
         'textarea',
-        '[tabindex]:not([tabindex="-1"]'
+        'button'
       ];
-      const focusableString = focusableElements.join(', ');
-      const modalEl = this.$refs.modalEl;
-      const focusables = modalEl.querySelectorAll(focusableString);
-      const firstFocusableElement = focusables[0];
-      const lastFocusableElement = focusables[focusables.length - 1];
 
-      modalEl.addEventListener('keydown', cycleFocusables);
+      const selector = elementSelectors
+        .map(addExcludingAttributes)
+        .join(', ');
 
-      firstFocusableElement.focus();
+      this.elementsToFocus = [ ...this.$refs.modalEl.querySelectorAll(selector) ]
```

Comment:
update the list of elements to focus, so that the `cycleElementsToFocus` listener from the mixin has the first and the last elements updated, which is necessary to trap the focus